File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int first[26] = {}, last[26] = {};
        for(int i = 1, j = s.size(); i <= s.size(); i++, j--){
            last[s[i-1] - 'a'] = i;
            first[s[j -1] - 'a'] = j;
        }
        
        int ans = INT_MAX;
        for(int i = 0; i < 26; i++)
            if(first[i] && first[i]==last[i])
                ans = min(ans, first[i]);
        
        return ans == INT_MAX? -1 : ans-1;
    }
};

File: 387-first-unique-character-in-a-string/387-first-unique-character-in-a-string.cs
public class Solution {
    public int FirstUniqChar(string s) {
        int[] cnt = new int[26];
        foreach(char c in s){
            cnt[c - 'a']++;
        }
        for(int i = 0; i < s.Length; i++)
            if(cnt[s[i] - 'a'] == 1)
                return i;
        return -1;
    }
}